File: python/problem1.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        length = len(nums);
        differDict = {};				#用字典存放差值--索引对
        for i in range(length):
            num = nums[i];
            item = differDict.get(num);	#看当前值是否已在差值--索引字典当中
            if item == None:			#如果不在的话，将目标值与当前值作为key，索引作为value
                differDict[target - num] = i;
            else:						#存在的话，直接返回
                return [item,i];

File: Java/problem1.java
public class Solution
{
    public int[] twoSum(int[] nums, int target) 
    {
        int[] result = new int[2];
        Map<Integer,Integer> map = new HashMap<Integer,Integer>();
        for(int i = 0;i < nums.length;++i)
        {
            if(map.containsKey(nums[i]))            //如果Map中已有当前值，说明找到了符合条件的一对数
            {
                result[0] = map.get(nums[i]);
                result[1] = i;
                return result;
            }
            map.put(target - nums[i],i);            //如果Map中并没有当前值，那就把目标值与当前值的差作为键，索引作为值
        }
        return result;
    }
}